lib/paperclip/storage/ftp/server.rb

```diff
@@ -58,12 +58,35 @@ def get_file(remote_file_path, local_file_path)
           connection.getbinaryfile(remote_file_path, local_file_path)
         end
 
-        def put_file(local_file_path, remote_file_path)
+        def put_file(local_file_path, remote_file_path, check_existence = true)
```

Comment:
I think the parameter name is misleading:
1) Since the method is called `put_file`, I would assume that the new parameter will check for a file existence, not a directory. 
2) It does not only check for existence, but also has the side effect of creating the directory.

So one option would be to just rename the parameter to e.g. `ensure_directory_exists`, but looking at the code we could also remove the parameter completely end let the caller create the directory via `mkdir_p` if needed to keep this method free from this side effect.

What do you think?